src/Symfony/Component/Security/Http/Firewall.php

```diff
@@ -78,6 +78,8 @@ public function onKernelFinishRequest(FinishRequestEvent $event)
     {
         $request = $event->getRequest();
 
+        $this->map->detachListeners($event->getRequest());
```

Comment:
This may not always work (think of other implementations of the `FirewallMapInterface`).